gapic-generator/templates/default/service/client/method/def/_options_defaults.erb

```diff
@@ -12,16 +12,19 @@ metadata[:"x-goog-api-client"] ||= Gapic::Headers.x_goog_api_client \
 <%- if method.routing_params? -%>
 
 header_params = {
-<%- method.routing_params.each do |rp| -%>
-  "<%= rp %>" => request.<%= rp %>,
+<%- routing_params = method.routing_params.dup -%>
+<%- last_routing_param = routing_params.pop -%>
+<%- routing_params.each do |routing_param| -%>
+  "<%= routing_param %>" => request.<%= routing_param %>,
 <%- end -%>
+  "<%= last_routing_param %>" => request.<%= last_routing_param %>
```

Comment:
This is really annoying to have to do in generated code. I'd support just setting `EnforcedStyleForMultiline: comma` in `Style/TrailingCommaInHashLiteral`.

Reply:
I would support an override to the default style for this _if_ there was a setting that allowed a trailing comma, but also allowed for the last element to not have a trailing comma. But AFAIK this rule does not allow for that type of configuration.

There are ways we can refactor this code to make it more declarative, like adding a method such as `Enumerable#each_with_last`. I would prefer to generate the code that satisfies the current Rubocop settings for this rule rather than change the rule.

Reply:
I've updated the code to be simpler in calculating whether to place a trailing comma.

Reply:
Or we could always do this...

```ruby
module Enumerable
  def each_with_comma
    return enum_for :each_with_comma unless block_given?
    
    each_with_object(count - 1).each_with_index.map do |(value, last), index| 
      comma = "," unless last == index
      yield [value, comma.to_s]
    end
  end
end
```

Reply:
Okay your updated implementation is a lot nicer. The only thing I'd change is to do this:
```ruby
comma = index == method.routing_params.count - 1 ? "" : ","
```

The reason is, the current implementation depends on `comma` being created as a new variable within the block each time. That's how it gets set to `nil` on the final iteration. But if at some point later, `comma` gets declared outside the block, then the final iteration will merely fail to update the variable, which will remain `","`. (Unfortunate side effect of Ruby not having explicit variable declaration.)

Oh and no, let's not add methods to `Enumerable`. :)